### PR TITLE
Feature/ab#131 calendar nightcare

### PR DESCRIPTION
**Pull request by BerendWouters**

### CODE REVIEW

File: Singer.API/ClientApp/src/app/modules/legalguardians/services/careRegistrationResult.dto.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export interface CareRegistrationResultDTO {
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/legalguardians/services/careRegistrationResult.dto.ts]: What's this DTO doing here? It looks like it lost its mommy, please explain it how to find its way to the DTOs folder ;)

File: Singer.API/ClientApp/src/app/modules/legalguardians/services/createCareRegistration.dto.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { EventRegistrationTypes } from '../shared/register-care-wizard/register-care-wizard.component';
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/legalguardians/services/createCareRegistration.dto.ts]: What's this DTO doing here? It looks like it lost its mommy, please explain it how to find its way to the DTOs folder ;)

File: Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/add-register-care-wizard-step.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export class AddRegisterCareWizardStep {
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/add-register-care-wizard-step.ts]: Why is this in a `shared` subfolder?

File: Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/register-care-wizard.component.css [hunk 4]
```diff
@@ -0,0 +1,201 @@
+/*
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/register-care-wizard.component.css]: Why is this in a `shared` subfolder?

File: Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/register-care-wizard.component.html [hunk 5]
```diff
@@ -0,0 +1,141 @@
+<mat-card class="card">
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/register-care-wizard.component.html]: Why is this in a `shared` subfolder?

File: Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/register-care-wizard.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/register-care-wizard.component.spec.ts]: Why is this in a `shared` subfolder?

File: Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/register-care-wizard.component.ts [hunk 7]
```diff
@@ -0,0 +1,212 @@
+import { Component, ViewChild } from '@angular/core';
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/legalguardians/shared/register-care-wizard/register-care-wizard.component.ts]: Why is this in a `shared` subfolder?

File: Singer.API/ClientApp/src/app/modules/legalguardians/shared/search-care-user-dialog/search-care-user-dialog.component.html [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<h2 mat-dialog-title>Zorggebruiker zoeken</h2>
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/legalguardians/shared/search-care-user-dialog/search-care-user-dialog.component.html]: Could this not have been a wrapper around the `CareUserSearchComponent` in `SharedModule`?

File: Singer.API/Controllers/CareRegistrationController.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            , createCareRegistration.CareUserIds, createCareRegistration.StartDateTime.LocalDateTime,
+            createCareRegistration.EndDateTime.LocalDateTime);
+         return new CareRegistrationResultDTO()
+         {
+            CreatedRegistrationIds = result
+         };
+      }
+   }
+
+   public class CreateCareRegistrationDTO
```
Comment (vyruz1986) [inline Singer.API/Controllers/CareRegistrationController.cs]: Another lost DTO 😲

File: Singer.API/Controllers/CareRegistrationController.cs [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+   {
+      public IReadOnlyList<Guid> CareUserIds { get; set; }
+      public DateTimeOffset StartDateTime { get; set; }
+      public DateTimeOffset EndDateTime { get; set; }
+      public EventRegistrationTypes EventRegistrationType { get; set; }
+   }
+
+   public class CareRegistrationResultDTO
```
Comment (vyruz1986) [inline Singer.API/Controllers/CareRegistrationController.cs]: Another lost DTO 😲

File: Singer.API/Controllers/CareRegistrationController.cs [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+      }
+
+      [HttpPost]
+      public async Task<ActionResult<CareRegistrationResultDTO>> Create(
+         [FromBody] CreateCareRegistrationDTO createCareRegistration)
+      {
+         var result = await _eventRegistrationService.Create(createCareRegistration.EventRegistrationType
+            , createCareRegistration.CareUserIds, createCareRegistration.StartDateTime.LocalDateTime,
+            createCareRegistration.EndDateTime.LocalDateTime);
+         return new CareRegistrationResultDTO()
```
Comment (vyruz1986) [inline Singer.API/Controllers/CareRegistrationController.cs:27]: Shouldn't this be returning a `Created(nameof(Get), ...` instead of a plain DTO?

**Review by vyruz1986 — CHANGES_REQUESTED** [review]

**Comment by vyruz1986** [comment]
* I am able to create a 'Dagopvang' registration with duration before 8AM and after 6PM, which should not be allowed.
* I can't find a way to approve/decline a request day/night care registration

File: Singer.API/ClientApp/src/app/modules/legalguardians/shared/search-care-user-dialog/search-care-user-dialog.component.html (hunk 8 above)
Comment (BerendWouters) [reply Singer.API/ClientApp/src/app/modules/legalguardians/shared/search-care-user-dialog/search-care-user-dialog.component.html]: Maybe, will resolve in the "ultimate-folder-cleanup"-ticket

File: Singer.API/Controllers/CareRegistrationController.cs (hunk 11 above)
Comment (vyruz1986) [reply Singer.API/Controllers/CareRegistrationController.cs:27]: @BerendWouters ? You resolved the comment without pushing a change or commenting

**Review by vyruz1986 — CHANGES_REQUESTED** [review]

File: Singer.API/Controllers/CareRegistrationController.cs (hunk 11 above)
Comment (BerendWouters) [reply Singer.API/Controllers/CareRegistrationController.cs:27]: Ah, yes.
~Will fix.~
Won't fix, since there's no route to return to after creation.

**Review by vyruz1986 — APPROVED** [review]

**Pull request merged**